superbench/benchmarks/micro_benchmarks/kernel_launch_overhead/CMakeLists.txt

```diff
@@ -2,10 +2,57 @@
 # Licensed under the MIT License.
 
 cmake_minimum_required(VERSION 3.18)
-project(kernel_launch_overhead LANGUAGES CUDA CXX)
 
-include(../cuda_common.cmake)
+project(kernel_launch_overhead LANGUAGES CXX)
+
+find_package(CUDAToolkit QUIET)
+if(CUDAToolkit_FOUND)
+    # Cuda environment
+    message(STATUS "Found CUDA: " ${CUDAToolkit_VERSION})
+    enable_language(CUDA)
+
+    include(../cuda_common.cmake)
+
+    add_executable(kernel_launch_overhead cuda_kernel_launch.cu)
+    set_property(TARGET kernel_launch_overhead PROPERTY CUDA_ARCHITECTURES ${NVCC_ARCHS_SUPPORTED})
+    install(TARGETS kernel_launch_overhead RUNTIME DESTINATION bin)
+else()
+    # ROCm environment
+    # Set ROCM_PATH
+    if(NOT DEFINED ENV{ROCM_PATH})
+    set(ROCM_PATH /opt/rocm)
+    else()
+    set(ROCM_PATH $ENV{ROCM_PATH})
+    endif()
+    # Set HIP_PATH
+    if(NOT DEFINED ENV{HIP_PATH})
+    set(HIP_PATH ${ROCM_PATH}/hip)
+    else()
+    set(HIP_PATH $ENV{HIP_PATH})
+    endif()
+    # Check if ROCm environment
+    if(EXISTS ${HIP_PATH})
+        # Search for hip in common locations
+        list(APPEND CMAKE_PREFIX_PATH ${HIP_PATH} ${ROCM_PATH})
+        set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
+        set(CMAKE_MODULE_PATH "${HIP_PATH}/lib/cmake/hip" ${CMAKE_MODULE_PATH})
```

Comment:
As Peng's comment, could we put ENV part into separate file for common usage?

Reply:
Sure